packages/fxa-auth-server/fxa-oauth-server/docs/prompt-none.md

```diff
@@ -0,0 +1,109 @@
+| `unauthorized_client`          | prompt=none is not enabled for the client       |
+| `* interaction_required`       | prompt=none cannot be used when requesting keys |
+| &nbsp;                         | OR account or session is not verified           |
+| `* login_required`             | user is not signed in                           |
+| `* account_selection_required` | a different user is signed in                   |
+
+`*` indicates the user must take some form of action to recover
+
+As an example, suppose the `authorization` request from the previous section failed because the user is not signed in and assume the `redirectURI` for client `ea3ca969f8c6bb0d` is `https://service.firefox.com/oauth/complete`. The user would
+be redirected to:
+
+```
+GET https://service.firefox.com/oauth/complete?state=2sfas415FSSF@A5f&error=login_required
+```
+
+### Recovering from errors
+
+At this point, the RP knows the user must authenticate to Firefox Accounts before OAuth codes or access tokens are returned. The RP can generate a new `state` and try again without the `prompt=none` query parameter:
+
+```
+GET https://accounts.firefox.com/authorization?client_id=ea3ca969f8c6bb0d&state=gASDF-3df@A5f&scope=profile
+```
+
+## Handling user logout
+
+Whenever the user terminates a session at the RP, any access and refresh tokens held by the RP for that session should be destroyed. Destroying the access and refresh tokens will ensure an entry for the session no longer appears in the [Devices and Apps][#fxa-devices-and-apps] list.
+
+### Destroying an access token
+
+```
+POST https://oauth.accounts.firefox.com/v1/destroy
+{
+  access_token: <access_token>
+}
+```
+
+### Destroying a refresh token
+
+```
+POST https://oauth.accounts.firefox.com/v1/destroy
+{
+  refresh_token: <refresh_token>
+}
+```
+
+More information on destroying tokens can be found on the [OAuth server's API docs][#oauth-server-api-destroy].
+
+### SSO logout
+
+[Several concerns are noted][#sso-logout], notably:
+
+> ...there's potentially confusing behaviour around logging out of an FxA relier. If I use FxA to sign in to AMO, and then I sign out in AMO, I could reasonably expect to have to enter my password again if I want to sign back in. But we don't have a way for AMO to signal back to FxA that the user signed out.
+
+This concern is not currently addressed as none of the [OIDC logout flows][#oidc-logout-flows] are currently supported.
+
+## prompt=none and security
+
+For users that are already authenticated to Firefox accounts, `prompt=none` bypasses the FxA authorization screen and redirects back to the RP without any user interaction. This behavior applies equally to users with 2FA enabled and could easily be misused by RPs to sign in users without the user realizing they had done so. The Firefox accounts team only enables the use of `prompt=none` for a service if both a good use case exists and the integration has been audited.
```

Comment:
We'll need to figure out what our guidelines for this auditing are. I'm not too worried about RPs abusing this for their own gain (that's what the OAuth consent model and our `trusted` flag are for) but I can imagine ways in which RPs might be tricked into doing a `prompt=none` flow without intending to.

For example, if you visit https://monitor.firefox.com/oauth/init?foo=bar right now, it will initiate an OAuth dance and blindly forward the `foo=bar` query parameters through to FxA. We probably don't want to pass through `prompt=none` is specified by some external caller.